src/Commands/Link.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Commands;
+
+use Exception;
+use Illuminate\Console\Command;
+use Illuminate\Support\LazyCollection;
+use Stancl\Tenancy\Actions\CreateStorageSymlinksAction;
+use Stancl\Tenancy\Actions\RemoveStorageSymlinksAction;
+use Stancl\Tenancy\Concerns\HasATenantsOption;
+
+class Link extends Command
+{
+    use HasATenantsOption;
+
+    /**
+     * The console command signature.
+     *
+     * @var string
+     */
+    protected $signature = 'tenants:link
+                {--tenants=* : The tenant(s) to run the command for. Default: all}
+                {--relative : Create the symbolic link using relative paths}
+                {--force : Recreate existing symbolic links}
+                {--remove : Remove symbolic links}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create or remove tenant symbolic links.';
+
+    /**
+     * Execute the console command.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $tenants = $this->getTenants();
+
+        try {
+            if ($this->option('remove')) {
+                $this->removeLinks($tenants);
+            } else {
+                $this->createLinks($tenants);
+            }
+        } catch (Exception $exception) {
+            $this->error($exception->getMessage());
+
+            return 1;
```

Comment:
> Can we return from this method? Are the returned values used?

I see the return value isn't used anywhere. I added this return value so we can test if the command failed using `->assertFailed()`. Now, the link command calls aren't tested. There are tests only for the jobs Create/RemoveSymlinks jobs. I think I should add these tests back, but I'd remove the `return 1` here and instead assert something else (e.g.  outputting an error message).

Also, the underlying actions for the jobs aren't separately tested because the jobs just do `XStorageSymlinksAction::handle($tenant)`, and the tests would look almost the same. It would probably be the best to have them tested either way.

@stancl, what do you think?

Reply:
> Now, the link command calls aren't tested

Why is that?

> I think I should add these tests back

Were they removed as part of some change?

Reply:
I removed them during a merge: https://github.com/archtechx/tenancy/pull/909/commits/8aff44215d920bed5adff810f6fe8836fbef9ad9#diff-2d46e8828aea0422777576ca8e903262bf80a262978ec02ad3d0783c38fe1b32L210-L215

I'm not sure but I think I removed them because I thought the job tests would be enough

Reply:
I think testing everything is best, if there's repetition I can show you how that can be fixed with [datasets](https://pestphp.com/docs/datasets) after I see the tests.

Reply:
I wrote the tests (https://github.com/archtechx/tenancy/blob/a08b0533a8fc354b644a74add9c725decb651ee5/tests/ActionTest.php). There's [a thing left to solve](https://github.com/archtechx/tenancy/pull/909#discussion_r959227902) though